jvb/src/main/kotlin/org/jitsi/videobridge/EndpointK.kt

```diff
@@ -261,6 +262,39 @@ class EndpointK @JvmOverloads constructor(
         return _transceiver.isReceivingVideo()
     }
 
+    /**
+     * Update media direction of {@link ChannelShim}s associated
+     * with this Endpoint.
+     *
+     * When media direction is set to 'sendrecv' JVB will
+     * accept incoming media from endpoint and forward it to
+     * other endpoints in a conference. Other endpoint's media
+     * will also be forwarded to current endpoint.
+     * When media direction is set to 'sendonly' JVB will
+     * NOT accept incoming media from this endpoint (not yet implemented), but
+     * media from other endpoints will be forwarded to this endpoint.
+     * When media direction is set to 'recvonly' JVB will
+     * accept incoming media from this endpoint, but will not forward
+     * other endpoint's media to this endpoint.
+     * When media direction is set to 'inactive' JVB will
+     * neither accept incoming media nor forward media from other endpoints.
+     *
+     * @param type media type.
+     * @param direction desired media direction:
+     *                       'sendrecv', 'sendonly', 'recvonly', 'inactive'
+     */
+    @SuppressWarnings("unused") // Used by plugins (Yuri)
+    override fun updateMediaDirection(type: MediaType, direction: String) {
+        when (direction) {
+            "sendrecv", "sendonly", "recvonly", "inactive" -> {
+                channelShims.firstOrNull { it.mediaType == type }?.setDirection(direction)
```

Comment:
I think this should be `foreach`, not `firstOrNull`, at least to have the same semantic as the Java code.

I think we never currently have more than one channel shim with the same media type, but if we ever did this code would only set the first one.

Reply:
Although that would match the old java behavior, I'm not sure the old behavior really makes sense.  I think we'd have a lot of broken code if we had more than one channel per media type, so I dunno that being flexible here adds much value.